### ISSUE TITLE
queries broken with version 2.0

### ISSUE THREAD

**Issue by duaneallam** [ryanfitz/vogels#93]
Hi 

The version of vogels in our project was automatically updated to version 2.0 and along with it all of our dyanamo queries broke.  Gets and indexes still work fine.  Downgrading to 2.0-rc8 temporarily fixes the problem.

here is an  example of a query we are using:

````
roleModel.query(user_oid).limit(100).attributes(['rid', 'name', 'parentRID']).exec(function (error, result) {
   ...
});
````
Which throws the following error
````
{ [ValidationException: Can not use both expression and non-expression parameters in the same request: Non-expression parameters: {AttributesToGet} Expression parameters: {KeyConditionExpression}]
  message: 'Can not use both expression and non-expression parameters in the same request: Non-expression parameters: {AttributesToGet} Expression parameters: {KeyConditionExpression}',
  code: 'ValidationException',
  time: Thu Nov 12 2015 04:29:33 GMT+0000 (UTC),
  statusCode: 400,
  retryable: false,
  retryDelay: 0 }
````

Any idea how this could be happening?

**Comment by CSharpFan** [comment]
Yup, same here. Can't find any tags though in this repo.

**Comment by ryanfitz** [comment]
This should be fixed in version 2.0.1. I had to switch to using the new DynamoDB expression syntax api, I didn't realize `attributes` was still using the legacy api. `attributes` now is using ProjectionExpression.

Let me know if you run into any other issues
